core/services/helper/main.py

```diff
@@ -35,16 +72,16 @@ def get_webpage_ports() -> List[int]:
         # And check if there is a webpage available that is not us
         ports = iter([connection.laddr.port for connection in connections])
         ports = filter(lambda port: port != PORT, ports)
-        ports = filter(Helper.check_webpage, ports)
-
-        return list(ports)
+        services = [Helper.check_port(port) for port in ports]
+        valid_services = [service for service in services if service.valid]
+        return valid_services
```

Comment:
You can do:
```python
services = map(Helper.check_port, ports)
valid_services = filter(lambda service: service.valid, services)
```

Also, check port is a terrible name, since it implies that will `check` or return a True and False and that is not the case.